tests/e2e/taqueria-plugin-flextesa.spec.ts

```diff
@@ -235,4 +237,231 @@ describe("E2E Testing for taqueria flextesa plugin",  () => {
+
+            // 2. Run taq typecheck hello-tacos.tz
+            const {stdout, stderr} = await exec(`taq typecheck ${dockerName} hello-tacos.tz`, {cwd: `./${taqueriaProjectPath}`});
+
+            // 3. Verify that it's well-typed and contains no errors
+            expect(stdout).toBe(contents.oneRowTable);
+            expect(stderr).toBe("");
+
+        } catch(error) {
+            throw new Error (`error: ${error}`);
+        }
+    });
+
+    test('Verify that taqueria typechecker task can typecheck all contracts under contracts folder', async () => {
+        try {
+            // 1. Copy two contracts from data folder to /contracts folder under taqueria project
+            execSync(`cp e2e/data/hello-tacos.tz ${taqueriaProjectPath}/contracts/hello-tacos-one.tz`);
+            execSync(`cp e2e/data/hello-tacos.tz ${taqueriaProjectPath}/contracts/hello-tacos-two.tz`);
+
+            // 2. Run taq typecheck
+            const {stdout, stderr} = await exec(`taq typecheck ${dockerName}`, {cwd: `./${taqueriaProjectPath}`});
+
+            // 3. Verify that both are well-typed and contain no errors
+            expect(stdout).toBe(contents.twoRowTable);
+            expect(stderr).toBe("");
+
+        } catch(error) {
+            throw new Error (`error: ${error}`);
+        }
+    });
+
+    test('Verify that taqueria typechecker task can typecheck multiple (but not all) contracts using typecheck [sourceFile1] [sourceFile2] command', async () => {
+        try {
+            // 1. Copy two contracts from data folder to /contracts folder under taqueria project
+            execSync(`cp e2e/data/hello-tacos.tz ${taqueriaProjectPath}/contracts/hello-tacos-one.tz`);
+            execSync(`cp e2e/data/hello-tacos.tz ${taqueriaProjectPath}/contracts/hello-tacos-two.tz`);
+            execSync(`cp e2e/data/hello-tacos.tz ${taqueriaProjectPath}/contracts/hello-tacos-three.tz`);
+
+            // 2. Run taq typecheck hello-tacos-one.tz hello-tacos-two.tz
+            const {stdout, stderr} = await exec(`taq typecheck ${dockerName} hello-tacos-one.tz hello-tacos-two.tz`, {cwd: `./${taqueriaProjectPath}`});
+
+            // 3. Verify that both are well-typed and contain no errors
+            expect(stdout).toBe(contents.twoRowTable);
+            expect(stderr).toBe("");
+
+        } catch(error) {
+            throw new Error (`error: ${error}`);
+        }
+    });
+
+    test('Verify that taqueria typechecker task will display proper message if user tries to typecheck contract that does not exist', async () => {
+        try {
+            // 1. Run taq typecheck ${contractName} for contract that does not exist
+            const {stdout, stderr} = await exec(`taq typecheck ${dockerName} test.tz`, {cwd: `./${taqueriaProjectPath}`})
+
+            // 2. Verify that output includes a table and an error message
+            expect(stdout).toBe(contents.nonExistent)
+            expect(stderr).toContain("Does not exist");
```

Comment:
So when it comes to the output of error conditions do we want to output both the error and the standard output? Is this case the table and a message are printed giving the user double of the information. It looks like the following:
```
test.tz: Does not exist

┌──────────┬────────────────┐
│        Contract     │             Result                   │
├──────────┼────────────────┤
│        test.tz         │         Does not exist          │
└──────────┴────────────────┘
```

Reply:
I think this is due to the same issue of outputting both stdout and stderr but wanted to add this output for context: 
```
taqueria/333 on  557-typechecker-task-JCC 
❯ taq simulate local taco.tz --storage '' -i ''  
 
Error:
  Erroneous command line argument 6 ().
  empty expression

Usage:
  tezos-client [global options] command [command options]
  tezos-client --help (for global options)
  tezos-client [global options] command --help (for command options)
  tezos-client --version (for version information)

To browse the documentation:
  tezos-client [global options] man (for a list of commands)
  tezos-client [global options] man -v 3 (for the full manual)

Global options (must come before the command):
  -d --base-dir <path>: client data directory (absent: TEZOS_CLIENT_DIR env)
  -c --config-file <path>: configuration file
  -t --timings: show RPC request times
  --chain <hash|tag>: chain on which to apply contextual commands (possible tags are 'main' and 'test')
  -b --block <hash|tag>: block on which to apply contextual commands (possible tags are 'head' and 'genesis')
  -w --wait <none|<int>>: how many confirmation blocks before to consider an operation as included
  -p --protocol <hash>: use commands of a specific protocol
  -l --log-requests: log all requests to the node
  -A --addr <IP addr|host>: [DEPRECATED: use --endpoint instead] IP address of the node
  -P --port <number>: [DEPRECATED: use --endpoint instead] RPC port of the node
  -S --tls: [DEPRECATED: use --endpoint instead] use TLS to connect to node.
  -m --media-type <json, binary, any or default>: Sets the "media-type" value for the "accept" header for RPC requests to the node. The media accept header indicates to the node which format of data serialisation is supported. Use the value "json" for serialisation to the JSON format.
  -E --endpoint <uri>: HTTP(S) endpoint of the node RPC interface; e.g. 'http://localhost:8732'
  -s --sources <path>: path to JSON file containing sources for --mode light. Example file content: {"min_agreement": 1.0, "uris": ["http://localhost:8732", "https://localhost:8733"]}
  -R --remote-signer <uri>: URI of the remote signer
  -f --password-filename <filename>: path to the password filename
  -M --mode <client|light|mockup|proxy>: how to interact with the node

┌──────────┬─────────┐
│ Contract │ Result  │
├──────────┼─────────┤
│ taco.tz  │ Invalid │
└──────────┴─────────┘
```

Reply:
I think it's not a big deal. But I can change the output inside the table from "Does not exist" to "N/A"? Would that be a better?

Reply:
It should not output any of the tezos-client specific information as the user is unable to use any of that error messaging through the abstraction

Reply:
Hmm I will see if I can trim it to just:

```
Error:
  Erroneous command line argument 6 ().
  empty expression
```

Reply:
@jchenche I think displaying the reason why a contract isn't valid makes sense. However, if there's an error due to incorrect usage of the tezos-client, then we shouldn't output the message from tezos-client but try to prevent the end-developer from incorrect invocation.